src/core/ext/filters/client_channel/lb_policy/pick_first/pick_first.cc

```diff
@@ -431,17 +448,20 @@ void PickFirst::PickFirstSubchannelData::ProcessConnectivityChangeLocked(
       size_t next_index =
           (sd->Index() + 1) % subchannel_list()->num_subchannels();
       sd = subchannel_list()->subchannel(next_index);
-      // Case 1: Only set state to TRANSIENT_FAILURE if we've tried
-      // all subchannels.
-      if (sd->Index() == 0 && subchannel_list() == p->subchannel_list_.get()) {
+      // If we're tried all subchannels, set state to TRANSIENT_FAILURE.
+      if (sd->Index() == 0) {
         p->channel_control_helper()->RequestReresolution();
```

Comment:
We should only re-resolve if this is the most recently created subchannel list, as discussed in https://github.com/grpc/grpc/pull/18021 https://reviewable.io/reviews/grpc/grpc/18021#-LZtoY2Q9PCo_dhxZNLm?

Reply:
Good point.  Done.